### ISSUE TITLE
error object on Query's children function is not reset

### ISSUE THREAD

**Issue by htulipe** [apollographql/react-apollo#2072]
I'm trying to handle error cases when a query variables mutates. My problem is with the error object on query's children function which is not reset when cache is read.

I have a Query component executing a query with a sort order variable. Something like:

```
query MyQuery(
  $order: ReviewSortOrder
) {
  reviews(
    order: $order
  ) {
    ...
  }
}
```
The ReviewSortOrder type is either `ASC` or `DESC` but right now only `ASC` is implemented and running the query with `DESC` returns an error from server.

Here is how my code executes:

- Default order is `ASC`: the query fires and gets a positive response. Query's children function is called with the data and **no error object**.
- User changes order via UI to `DESC`: the query fires again and gets an error response. Query's children function is called **with an error object**.
- User changes back to `ASC`: the query does not fire because the cache is used. Query's children function is called **with the previous error object**.

**Intended outcome:**

I'd expect the last call to children to not include the error object.

**Actual outcome:**

The last call to children contains the error object from the previous execution

**How to reproduce the issue:**

I don't have a reproduction repo or sandbox, sorry. My Query is configured with `fetchPolicy` at `cache-first` and `notifyOnNetworkStatusChange` at `true`, other props use their default values.

**Version**

* apollo-client@2.2.8
* react-apollo@2.1.3

Many thanks for your help

**Comment by hughfm** [comment]
Interested if anyone has a reasonable way to handle this behavior? Is it considered a bug or expected?

**Comment by hwillson** [comment]
This was addressed in https://github.com/apollographql/react-apollo/pull/2718, which was included in `react-apollo` 2.5.3. Thanks!
